omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/AdmissionLocationResource.java

```diff
@@ -61,7 +61,12 @@ public class AdmissionLocationResource extends DelegatingCrudResource<AdmissionL
 	@Override
 	protected PageableResult doGetAll(RequestContext context) throws ResponseException {
 		List<AdmissionLocation> admissionLocations = Context.getService(BedManagementService.class).getAdmissionLocations();
-		return new AlreadyPaged<AdmissionLocation>(context, admissionLocations, false);
+		return new NeedsPaging<>(admissionLocations, context);
+	}
+	
+	@Override
+	protected PageableResult doSearch(RequestContext context) throws ResponseException {
+		return doGetAll(context);
```

Comment:
This exposes a (likely) overall bug in the REST module, whereas if I try to hit a URL like:

```http://localhost:8080/openmrs/ws/rest/v1/admissionLocation?totalCount=true```

Then the presence of the `totalCount` parameter seems to activate the `doSearch` method rather than the `doGetAll` method, and the default implementation of `doSearch` in the `DelegatingCrudResource` superclass is:

```
    protected PageableResult doSearch(RequestContext context) {
        return new EmptySearchResult();
    }
```
So, without adding this implementation, hitting http://localhost:8080/openmrs/ws/rest/v1/admissionLocation results in, say, 20 results, where as hitting http://localhost:8080/openmrs/ws/rest/v1/admissionLocation?totalCount=true results in 0 results.